README.md

```diff
@@ -1,26 +1,81 @@
 # docker-timemachine
 A docker container to compile the lastest version of Netatalk in order to run a Time Machine server.
 
+
 ## Installation
 
+### Step 1 - Start the Server
+
 To download the docker container and execute it, simply run:
 
 ```
-$ docker run -h timemachine --name timemachine --restart=unless-stopped -d -v /route/to/your/timemachine:/timemachine -it -p 548:548 -p 636:636 odarriba/timemachine
+$ docker run -h timemachine --name timemachine --restart=unless-stopped -d -v /external_volume:/timemachine -it -p 548:548 -p 636:636 odarriba/timemachine
 ```
 
-To add users, just run this command:
+Replace `external_volume` with a local path where you want to store your data.
+
+As the image has been started using the `--restart=always` flag, it will start when the computers boots up.
+
+
+
+### Step 2 - Add a User
+
+To add a user, run:
 
 ```
 $ docker exec timemachine add-account USERNAME PASSWORD MOUNT_POINT [VOL_SIZE_MB]
 ```
 
 But take care that:
-* `MOUNT_POINT` should be an absolute path, preferably inside `/timemachine`, so it will be stored in your external volume.
+* `MOUNT_POINT` should be an absolute path, preferably set it to `"/timemachine"`, so it will be stored right in your external volume.
 * `VOL_SIZE_MB` is an optional parameter. It indicates the max volume size for that user.
 
 Now you have a docker instance running `netatalk`.
 
+
+### Step 3 - Enable Auto Discovery
+
+Avahi daemon is commonly used to help your computers to find the services provided by a server.
+
+Avahi isn't built into this Docker image because, due to Docker's networking limitations, Avahi can't spread it's messages to announce the services.
+
+**If you want to enable this feature, you can install Avahi daemon on your host** following these steps (Ubuntu version):
+
+* Install `avahi-daemon`: run `sudo apt-get install avahi-daemon avahi-utils`
+* Copy the file from `avahi/nsswitch.conf` to `/etc/nsswitch.conf`
+* Copy the service description file from `avahi/afpd.service` to `/etc/avahi/services/afpd.service`
+* Restart Avahi's daemon: `sudo /etc/init.d/avahi-daemon restart`
+
+
+### Step 4 - Configure Your Firewall
+
+Make sure
+
+* your server can receive traffic on port `548` and `636` (e.g., `ufw allow 548`, (`636` respectively)).
+
+* your Mac allows outgoing connections (Little Snitch?)
+
+
+
+### Step 5 - Start Using It
+
+To start using it, follow these steps:
+
+* If you use Avahi, open **Finder**, go to **Shared** and connect to your server with your new username and password. Alternatively, or if you don't use Avahi, from **Finder** press **CMD-K** and type `afp://your-server-name`.
```

Comment:
Maybe here you can add that, if there is any problem with auto-discovering within the network, you can mount the volume using `afp://your-server-ip-address`.